src/sql/workbench/browser/modal/calloutDialog.ts

```diff
@@ -0,0 +1,329 @@
+
+	constructor(
+		calloutType: CalloutType,
+		title: string,
+		triggerProperties: ITriggerProperties,
+		@IPathService private readonly _pathService: IPathService,
+		@IFileDialogService private readonly _fileDialogService: IFileDialogService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IContextViewService private _contextViewService: IContextViewService,
+		@IClipboardService clipboardService: IClipboardService,
+		@ILogService logService: ILogService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService
+	) {
+		super(
+			title,
+			TelemetryKeys.CalloutDialog,
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			{
+				dialogStyle: 'callout',
+				dialogPosition: 'below',
+				triggerProperties: triggerProperties
+			});
+
+		this._selectionComplete = new Deferred<ICalloutDialogOptions>();
+		this._calloutType = calloutType;
+	}
+
+	/**
+	 * Opens the dialog and returns a promise for what options the user chooses.
+	 */
+	public open(): Promise<ICalloutDialogOptions> {
+		this.show();
+		return this._selectionComplete.promise;
+	}
+
+	public render() {
+		super.render();
+
+		attachModalDialogStyler(this, this._themeService);
+
+		this._insertButton = this.addFooterButton(this.insertButtonText, () => this.insert());
+		attachButtonStyler(this._insertButton, this._themeService);
+
+		this._cancelButton = this.addFooterButton(this.cancelButtonText, () => this.cancel(), null, true);
+		attachButtonStyler(this._cancelButton, this._themeService);
+
+		this.registerListeners();
+	}
+
+	protected renderBody(container: HTMLElement) {
+		if (this._calloutType === 'IMAGE') {
+			this.buildInsertImageCallout(container);
+		}
+
+		if (this._calloutType === 'LINK') {
+			this.buildInsertLinkCallout(container);
+		}
+	}
+
+	private buildInsertImageCallout(container: HTMLElement): void {
+		let imageContentColumn = DOM.$('.column.insert-image');
+		DOM.append(container, imageContentColumn);
+
+		let locationRow = DOM.$('.row');
+		DOM.append(imageContentColumn, locationRow);
+
+		this._imageLocationLabel = DOM.$('p');
+		this._imageLocationLabel.innerText = this.locationLabel;
+		DOM.append(locationRow, this._imageLocationLabel);
+
+		let radioButtonGroup = DOM.$('.radio-group');
+		this._imageLocalRadioButton = new RadioButton(radioButtonGroup, {
+			label: this.localImageLabel,
+			enabled: true,
+			checked: true
+		});
+		this._imageRemoteRadioButton = new RadioButton(radioButtonGroup, {
+			label: this.remoteImageLabel,
+			enabled: true,
+			checked: false
+		});
+		this._imageLocalRadioButton.value = localize('local', "Local");
+		this._imageLocalRadioButton.name = 'group1';
```

Comment:
Use a better group name please - our button logic currently looks at all names for any radio buttons and so having a common name like this increases the chance of there being collisions.

Reply:
and the name should be a const defined somewhere so it's easier to update and not miss any.

Reply:
These two radio buttons are defined on line 51. Is that not sufficient? If you really want them declared as const, I can do so, at the start of the class.

Reply:
Updated.